apps/MCF/mcf_sparse_matrix.cuh

```diff
@@ -49,4 +170,102 @@ __global__ static void mcf_A_B_setup(
     query.dispatch<Op::VV>(block, shrd_alloc, init_lambda);
 }
 
-void mcf_rxmesh_solver() {}
\ No newline at end of file
+template <typename T>
+void mcf_rxmesh_cusolver_chol(rxmesh::RXMeshStatic&              rxmesh,
+                              const std::vector<std::vector<T>>& ground_truth)
+{
+    using namespace rxmesh;
+    constexpr uint32_t blockThreads = 256;
+
+    uint32_t num_vertices = rxmesh.get_num_vertices();
+    auto     coords       = rxmesh.get_input_vertex_coordinates();
+
+    SparseMatrix<float> A_mat(rxmesh);
+    DenseMatrix<float>  X_mat(num_vertices, 3);
+    DenseMatrix<float>  B_mat(num_vertices, 3);
+
+    printf("use_uniform_laplace: %d, time_step: %f\n",
+           Arg.use_uniform_laplace,
+           Arg.time_step);
+
+    // B set up
+    LaunchBox<blockThreads> launch_box_B;
+    rxmesh.prepare_launch_box({Op::VV},
+                              launch_box_B,
+                              (void*)mcf_B_setup<float, blockThreads>,
+                              !Arg.use_uniform_laplace);
+
+    mcf_B_setup<float, blockThreads><<<launch_box_B.blocks,
+                                       launch_box_B.num_threads,
+                                       launch_box_B.smem_bytes_dyn>>>(
+        rxmesh.get_context(), *coords, B_mat, Arg.use_uniform_laplace);
+
+    CUDA_ERROR(cudaDeviceSynchronize());
+
+    // A and X set up
+    LaunchBox<blockThreads> launch_box_A_X;
+    rxmesh.prepare_launch_box({Op::VV},
+                              launch_box_A_X,
+                              (void*)mcf_A_X_setup<float, blockThreads>,
+                              !Arg.use_uniform_laplace);
+
+    mcf_A_X_setup<float, blockThreads>
+        <<<launch_box_A_X.blocks,
+           launch_box_A_X.num_threads,
+           launch_box_A_X.smem_bytes_dyn>>>(rxmesh.get_context(),
+                                            *coords,
+                                            A_mat,
+                                            X_mat,
+                                            Arg.use_uniform_laplace,
+                                            Arg.time_step);
+
+    // Solving the linear system using chol factorization and no reordering
+    A_mat.spmat_linear_solve(B_mat, X_mat, Solver::CHOL, Reorder::NONE);
+
+    auto smooth_X =
+        rxmesh.add_vertex_attribute<T>("smooth_X", 3, rxmesh::LOCATION_ALL);
+    auto truth_X =
+        rxmesh.add_vertex_attribute<T>("truth_X", 3, rxmesh::LOCATION_ALL);
+
+    LaunchBox<blockThreads> launch_box_smooth;
+    rxmesh.prepare_launch_box({Op::VV},
+                              launch_box_smooth,
+                              (void*)update_smooth_result<float, blockThreads>);
+
+    update_smooth_result<float, blockThreads>
+        <<<launch_box_smooth.blocks,
+           launch_box_smooth.num_threads,
+           launch_box_smooth.smem_bytes_dyn>>>(
+            rxmesh.get_context(), *smooth_X, A_mat, X_mat);
+    smooth_X->move(rxmesh::DEVICE, rxmesh::HOST);
```

Comment:
I think there are three solutions that we could implement that to avoid having this kernel 
1. Implement a `move` function in the dense matrix that moves the data from a dense matrix to an attribute 
2. Use `for_each_vertex()` function from `RXMeshStatic` to implement this in the application since we only need an access to the vertex handle (not query needed) 
3. Implement an overloading to `operator()` that takes a `VertexHandle` and use it to find the linear ID. This will require storing a shared pointer to an instance of RXMeshStatic inside the dense matrix.

I would vote for 3 since it avoids allocating another attribute. So, the user would only need to call a `move(DEVICE, HOST)` which move the dense matrix from device to the host and then they could access the dense matrix directly using the handle.